cql3/restrictions/statement_restrictions.cc

```diff
@@ -450,6 +450,60 @@ statement_restrictions::statement_restrictions(data_dictionary::database db,
     if (_uses_secondary_indexing && !(for_view || allow_filtering)) {
         validate_secondary_index_selections(selects_only_static_columns);
     }
+
+    calculate_column_defs_for_filtering_and_erase_restrictions_used_for_index(db);
+
+    if (pk_restrictions_need_filtering()) {
+        auto partition_key_filter = expr::conjunction{
+            .children = _single_column_partition_key_restrictions
+                    | std::ranges::views::values
+                    | std::ranges::to<std::vector>(),
+        };
+        _partition_level_filter = expr::make_conjunction(std::move(_partition_level_filter), std::move(partition_key_filter));
+    }
+
+    if (ck_restrictions_need_filtering()) {
+        auto clustering_key_filter = expr::conjunction{
+            .children = _single_column_clustering_key_restrictions
+                    | std::ranges::views::values
+                    | std::ranges::to<std::vector>(),
+        };
+        _clustering_row_level_filter = expr::make_conjunction(std::move(_clustering_row_level_filter), std::move(clustering_key_filter));
+    }
+
+    auto check_column_kind = [] (column_kind kind, const expr::single_column_restrictions_map::value_type& v) -> bool {
+        return v.first->kind == kind;
+    };
+
+    auto make_column_kind_checker = [&] (column_kind kind) {
+        return std::bind_front(check_column_kind, kind);
+    };
+
+    auto static_columns_filter = expr::conjunction{
+        .children = _single_column_nonprimary_key_restrictions
+                | std::ranges::views::filter(make_column_kind_checker(column_kind::static_column))
+                | std::ranges::views::values
+                | std::ranges::to<std::vector>(),
+    };
+
+    _partition_level_filter = expr::make_conjunction(std::move(_partition_level_filter), std::move(static_columns_filter));
+
+    auto regular_columns_filter = expr::conjunction{
+        .children = _single_column_nonprimary_key_restrictions
+                | std::ranges::views::filter(make_column_kind_checker(column_kind::regular_column))
+                | std::ranges::views::values
+                | std::ranges::to<std::vector>(),
```

Comment:
I consider this accidental duplication. It's not a duplicate because of some deep invariant, but because of how the code happens to be structured. My expectation is that we'll lose the symmetry between static and regular rows soon.

One example of how they're asymmetric is `WHERE f(static_col, regular_col)` (not legal now). This will have to get pulled towards the regular column side, because it can't be evaluated once per partition like static columns.

I expect the code to evolve from filtering towards partitioning, where we partition the boolean factors of the WHERE clause to a more-optimized-part and a less-optimized-part.